Formula/liquidctl.rb

```diff
@@ -0,0 +1,46 @@
+class Liquidctl < Formula
+  include Language::Python::Virtualenv
+
+  desc "Cross-platform tool and drivers for liquid coolers and other devices"
+  homepage "https://github.com/jonasmalacofilho/liquidctl"
+  url "https://files.pythonhosted.org/packages/source/l/liquidctl/liquidctl-1.3.0.tar.gz"
+  sha256 "ce0483b0a7f9cf2618cb30bdf3ff4195e20d9df6c615f69afe127f54956e42ce"
+
+  head "https://github.com/jonasmalacofilho/liquidctl.git"
+
+  depends_on "libusb"
+  depends_on "python"
+
+  resource "docopt" do
+    url "https://files.pythonhosted.org/packages/source/d/docopt/docopt-0.6.2.tar.gz"
+    sha256 "49b3a825280bd66b3aa83585ef59c4a8c82f2c8a522dbe754a8bc8d08c85c491"
+  end
+
+  resource "hidapi" do
+    url "https://files.pythonhosted.org/packages/source/h/hidapi/hidapi-0.7.99.post21.tar.gz"
+    sha256 "e0be1aa6566979266a8fc845ab0e18613f4918cf2c977fe67050f5dc7e2a9a97"
+  end
+
+  resource "pyusb" do
+    url "https://files.pythonhosted.org/packages/source/p/pyusb/pyusb-1.0.2.tar.gz"
+    sha256 "4e9b72cc4a4205ca64fbf1f3fff39a335512166c151ad103e55c8223ac147362"
+  end
+
+  def install
+    # customize liquidctl --version
+    ENV["DIST_NAME"] = "homebrew"
+    ENV["DIST_PACKAGE"] = "#{tap.nil? ? path : full_name} #{version}"
```

Comment:
Why is this needed?

Reply:
Upstream developer here...

This customizes the output of `liquidctl --version` adding some downstream package information so I know that I can (sort of) trust the version information and track down any build/dependency/platform issues.

It's by no means required, but it's nice to have (for me).

Reply:
And the conditional is needed so the formula can still be tested locally/directly from the file.

```rb
# The {Tap} instance associated with this {Formula}.
# If it's <code>nil</code>, then this formula is loaded from path or URL.
# @private
attr_reader :tap
```

https://github.com/Homebrew/brew/blob/d65ad9fe1a96fd895d24e097a3d69b9d280ee902/Library/Homebrew/formula.rb#L96-L99

Reply:
That seems like a lot of code that only happens when you write the formula the first time.

Reply:
I'm not sure, but wouldn't the formula be installed from the local file when testing an updated version?

Anyway, this is your area of expertise and

```
ENV["DIST_PACKAGE"] = "#{full_name} #{version}"
```

would be fine be me.

Reply:
@SMillerDev, you never answered: wouldn't a user trying to update _and test_ the formula need to install it directly from the local file?

@CaseySJ feel free to remove DIST_NAME/DIST_PACKAGE.

Reply:
No, actually you don't even need to for writing it the first time. You can just edit the file in `$(brew --prefix homebrew/core)/Formulae/liquidctl.rb` and test it using `brew test liquidctl`.

Reply:
@SMillerDev oh, good to know, thanks!

So would

```
    # customize liquidctl --version
    ENV["DIST_NAME"] = "homebrew"
    ENV["DIST_PACKAGE"] = "liquidctl #{version}"
```

be acceptable or should Casey remove everything?

Reply:
This makes sense to keep. Although the package might be a little redundant? The liquidctl package will always be liquidctl?

Reply:
The idea is that `DIST_PACKAGE` should match the user visible name of the actual package.  So, _in homebrew-core,_ it should always be `liquidctl #{version}`.

But, for comparison, in the custom tap I use `#{full_name} #{version}`, and on Linux distros this can vary wildly and include other important information like build version.

---

On liquidctl side...

By default `liquidctl --version` will only output the base/hardcoded version (e.g. `liquidctl v1.3.1`).

But if both `DIST_NAME` and `DIST_PACKAGE` are passed, the output will then include this additional information.  An example would be: `liquidctl v1.3.1 (Arch Linux; python-liquidctl-git 1.3.1.r1.8e67646-1)`.

If manually installed from sources, with git, it will include the commit hash.  And if installed in pip --editable mode, it will instead just make a note of that.

Reply:
@CaseySJ I think Sean meant that the cleaner `ENV["DIST_PACKAGE"] = "liquidctl #{version}"` made sense to keep.

Reply:
Understood. Will update the formula this evening and push the branch again.